### PR TITLE
Deprecate Bukkit#getLogger

### PR DESCRIPTION
**Pull request by powercasgamer**
https://discord.com/channels/289587909051416579/925530366192779286/1225482929333997609

### CODE REVIEW

**Comment by mbax** [comment]
Perhaps note in the deprecation message *why* this is a bad idea to use?

**Comment by MiniDigger** [comment]
> I'd propose additionally wiring it into the SysoutCatcher or similar, so that it nags authors to stop using it. As demonstrated by the SysoutCatcher, an effective way to get plugin authors to stop using something is to get half their userbase to nag them about it.

not instantly, that would be meh, at least give people one cycle to notice the deprecation

**Comment by powercasgamer** [comment]
> Perhaps note in the deprecation message _why_ this is a bad idea to use?

Do you have a preferred message?

**Comment by Lulu13022002** [comment]
This could be merged in the patch "Fix Spigot annotation mistakes"

File: patches/api/0468-Deprecate-Bukkit-getLogger.patch [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+diff --git a/src/main/java/org/bukkit/Bukkit.java b/src/main/java/org/bukkit/Bukkit.java
+index db46e3233edf18bd91a6da8612ccc9f0080d5c17..ab532ab39a373cc0e832e23528eb95597f66f334 100644
+--- a/src/main/java/org/bukkit/Bukkit.java
++++ b/src/main/java/org/bukkit/Bukkit.java
+@@ -1056,8 +1056,10 @@ public final class Bukkit {
+      * Returns the primary logger associated with this server instance.
+      *
+      * @return Logger associated with this server
++     * @deprecated This logger doesn't have a plugin prefix and therefore makes it hard for people to debug. Use {@link org.bukkit.plugin.Plugin#getLogger()}
```
Comment (electronicboy) [inline patches/api/0468-Deprecate-Bukkit-getLogger.patch]: Somebody who can tonguetangulate better than me would be appreciated, but, generally;
This logger is owned by the Minecraft server software, for logging plugin messages, you should use a Logger from within your own contexts, i.e. Plugin#getLogger(), or create a logger for your Class (i.e. slf4j) in order to retain contextual information which is used for identifying the source of messages

**Review by electronicboy — APPROVED** [review]

**Review by lynxplay — APPROVED** [review]

**Pull request merged**